test/test_package_version.py

```diff
@@ -61,55 +63,57 @@ def test_update_versions(self):
         finally:
             shutil.rmtree(root_dir)
 
-    def test_update_changelog_unicode(self):
+    @in_temporary_directory
+    def test_update_changelog_unicode(self, directory=None):
         """Test that updating the changelog does not throw an exception on unicode characters."""
-        with tempfile.NamedTemporaryFile() as temp_file:
-            missing_changelogs_but_forthcoming = {}
-            # Mock the Changelog object from catkin_pkg
-            mock_changelog = mock.Mock()
-            # Create a changelog entry with a unicode char.
-            mock_changelog.rst = ('^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n'
-                                  'Changelog for package fake_pkg\n'
-                                  '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n'
-                                  '\n'
-                                  'Forthcoming\n'
-                                  '-----------\n'
-                                  '* This is my changelog entry\n'
-                                  '* This is a line that has unicode' u'\xfc''\n'
-                                  '\n'
-                                  '0.0.9 (2017-01-30)\n'
-                                  '------------------\n'
-                                  '* This is old version.\n')
+        fd, temp_file = tempfile.mkstemp(dir=directory)
+        os.close(fd)
```

Comment:
I don't think the `temp_file` needs to be created here. I is being written to by the called functions and later read back. A simple patch should be sufficient here:

```
temp_file = os.path.join(directory, 'changelog')
```